### ISSUE TITLE
Accessibility: Cannot open form tabs with keyboard

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-admin#1966]
WCAG 2.1.1

The interactivity for tabs in a form is incorrect.
https://www.w3.org/WAI/ARIA/apg/patterns/tabs/ provides correct keyboard interaction for tabs.

We correctly use home/end/tab/arrow keys, but there is currently no way to actually open a tab panel. Pressing either space or enter should open the tab panel - or else the tab panel should open automatically when its tab is given focus (the latter is recommended under certain circumstances, see link above).

## Notes

- Tab panels _can_ be made to lazy-load in the CMS, which makes it hard to know whether we should follow the recommendation that "tabs activate automatically when they receive focus as long as their associated tab panels are displayed without noticeable latency". We should probably just require space or enter to be pressed across the board.

## Acceptance criteria

- [x] Form tab panels can be opened using the keyboard
- [x] Opening a tab panel _doesn't_ change where the focus is automatically
- [x] Pressing `tab` after opening a tab takes you to the first focusable element inside the tab panel
- [ ] If the tab panel has no focusable elements inside it, the tab panel itself should have `tabindex="0"` (if feasible). update - created new [issue](https://github.com/silverstripe/silverstripe-admin/issues/2014) for this
- [x] Check both entwine and react forms

## Kitchen sink CI
- https://github.com/creative-commoners/recipe-kitchen-sink/actions/runs/18360720482

## PRs
- https://github.com/silverstripe/silverstripe-admin/pull/2012
- https://github.com/silverstripe/silverstripe-asset-admin/pull/1616
- https://github.com/silverstripe/developer-docs/pull/824

**Comment by emteknetnz** [comment]
> Opening a tab panel doesn't change where the focus is automatically

There is an existing pattern where using left/right arrow keys on tab will automatically 'click' on the tab - this is done by jquery-ui tab https://jqueryui.com/tabs/ - at least for the secondary/panel navigation, and the primary nav for some things such as regular dataobjects

Because of this we should automatically "click on" the tabs as they are focused.

There is an existing issue where the some of the tabs don't do this:
- The primary nav for pages does not - due to this code https://github.com/silverstripe/silverstripe-admin/blob/3.0/client/src/legacy/LeftAndMain.js#L1434 - I tried removing this code but then AJAX just breaks, so I simply added new code in this PR to fix this
- React tabs weren't even tabbable. I've made them tabbable and the left/right arrows now behave like the jquery tabs

**Comment by emteknetnz** [comment]
> If the tab panel has no focusable elements inside it, the tab panel itself should have tabindex="0" (if feasible).

I don't this this AC belongs in this issue
- If there are no inputs in the form, then that's really a developer issue, what is tabbing to the form even supposed to do?
- If there are input and they are all in a read-only state and not tabbable e.g. they all render as `<p>` tags or something like that, then probably the solution is to add tabIndex="0" to all of those and maybe some aria tags so that they're read out. Though this should be done is a different issue.

**Comment by GuySartorelli** [comment]
> > If the tab panel has no focusable elements inside it, the tab panel itself should have tabindex="0" (if feasible).
> 
> I don't this this AC belongs in this issue
> - If there are no inputs in the form, then that's really a developer issue, what is tabbing to the form even supposed to do?
> - If there are input and they are all in a read-only state and not tabbable e.g. they all render as `<p>` tags or something like that, then probably the solution is to add tabIndex="0" to all of those and maybe some aria tags so that they're read out. Though this should be done is a different issue.

This acceptance criteria was added because it's specifically mentioned in the [W3C tabs pattern document](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/). It's the recommended way for the tab to function.

> If there are no inputs in the form, then that's really a developer issue

If that's the case, an error or warning should be created. But we don't do that and IMO we shouldn't do that. Having an empty tab (or rather, a tab with no interactive elements) is weird, but it's valid. Maybe it has a massive chunk of text info or something, who knows.

> If there are input and they are all in a read-only state and not tabbable

Read-only form fields should be tabbable

> probably the solution is to add tabIndex="0" to all of those

If they're not interactive elements, we shouldn't add them to the tab order. The recommendation from W3C is to make the tab container itself tabbable. We should follow that suggestion (which will be a common pattern across the web) instead of guessing and making it up ourselves.

**Comment by GuySartorelli** [comment]
@emteknetnz Please respond to the comment above.

**Comment by emteknetnz** [comment]
OK so you're referring to `When the tabpanel does not contain any focusable elements or the first element with content is not focusable, the tabpanel should set tabindex="0" to include it in the tab sequence of the page` 

The AC has `(if feasible).` - I'd say this is going to be a significant amount of effort due to the 3x types of panels we have - primary tabs that pjaxes in heaps of content, secondary nav that shows/hides content based using jquery, react based tabs that do their own thing. I'll put this AC on to its own [card](https://github.com/silverstripe/silverstripe-admin/issues/2014).

**Comment by GuySartorelli** [comment]
PRs merged
